pkg/client/cache/listers.go

```diff
@@ -111,6 +111,44 @@ func (s storePodsNamespacer) List(selector labels.Selector) (api.PodList, error)
 	return pods, nil
 }
 
+// Please note that selector is filtering among the pods that have gotten into
+// the store; there may have been some filtering that already happened before
+// that.
+func (s storePodsNamespacer) ListWithPredicate(pickPod func(*api.Pod) bool) (api.PodList, error) {
```

Comment:
Oh sorry, this piece of code is actually dead. I ended up listing all pods and go over all pods to find the ones I need ([here](https://github.com/kubernetes/kubernetes/pull/27600/files/eb66453972b391aef4217f46594ca6a34e3d2cbf#diff-d79787baf9b2ad960d49b76e915c13dfR697)). And I see what you are saying. I'll remove this piece of code.